sections/advanced/reaching-ui-state.md

```diff
@@ -0,0 +1,222 @@
+// all calls to the network route will be stubbed by the data in agents.json file
+```
+
+#### What if there are so many network requests happening?
+Where do we get all our mocks for fixtures? We do not want to manually copy and save them. We want to record them as the test runs against a real API
+
+There are at least 2 Cypress plugins you can utilize for this [1](https://github.com/Nanciee/cypress-autorecord) & [2](https://github.com/scottschafer/cypressautomocker).
+
+If these do not work for you, you can easily create record and playback utilities yourself with these 3 functions.
+
+```javascript
+function stubRecorder(pathToJson) {
+  const xhrData = []; // an empty array to hold the data
+  cy.server({ // if recording, save the response data in the array
+    onResponse: (response) => {
+      const url = response.url;
+      const method = response.method;
+      const data = response.response.body;
+      // We push a new entry into the xhrData array
+      xhrData.push({ url, method, data });
+    }
+  });
+  // this is used as a ‘filter’ for the data you want to record.
+  // You can specify the methods and routes that are recorded
+  cy.log('recording!');
+  cy.route({
+    method: 'GET',
+    url: '*',
+  });
+  // if recording, after the test runs, create a fixture file with the recorded data
+  after(function () {
+    cy.writeFile(`./cypress/fixtures/${pathToJson}.json`, xhrData);
+    cy.log(`Wrote ${xhrData.length} XHR responses to local file ${pathToJson}.json`);
+  });
+}
+
+/** Plays recorded fixture with all required network data as json*/
+function playStubbedFixture(stateFixture) {
+  cy.log(`playing fixture from ${stateFixture}`);
+  cy.fixture(stateFixture, { timeout: 15000 }) // the fixture file may be large and take time in CI
+    .then(data => {
+      for (let i = 0, length = data.length; i < length; i++) {
+        cy.route(data[i].method, data[i].url, data[i].data);
+      }
+    }).as(`stateFixture_stub`);
+}
+
+/** Visits the stubbed state */
+function visitStubbedState(stubFile, url, wait: boolean = true) {
+  playStubbedFixture(stubFile);
+  cy.visit(url);
+  if (wait) { // sometimes you do not want to wait for network, this gives you the option
+    cy.wait('@stateFixture_stub', { timeout: 15000 });
+  }
+}
+
+//////////
+// usage
+
+// recording network
+it('should run your test', function () {
+  stubrecorder('jsonfileNameForNetworkData');
+
+  // your original test
+
+  cy.wait(5000); // one-time wait so that the after() step records all the network without missing anything
+
+  // the rest of your original test
+});
+
+// playing the stubbed network
+it('should run your test', function () {
+  // every time we visit this endpoint, all network will be stubbed
+  // double check this by observing (XHR stubbed) network responses in the test runner
+  visitStubbedState('jsonfileNameForNetworkData', '/endpoint');
+
+  // the rest of your original test
+});
+```
+
+
+<br/><br/>
+
+### Application actions
```

Comment:
The App Actions section should stay before the stub recording one because the latter is a more advanced

Reply:
moved them around